hackathon_site/dashboard/frontend/src/components/general/Header/Header.js

```diff
@@ -5,39 +5,71 @@ import AppBar from "@material-ui/core/AppBar";
-                    <Logo
-                        className={styles.headerLogoImg}
-                        alt="Hackathon logo"
-                        data-testid="headerLogo"
-                    />
-                    <Typography variant="h6" data-testid="hackathonName">
-                        Hackathon Name
-                    </Typography>
-                </div>
-            </Hidden>
-            <Hidden implementation="css" mdUp>
-                {showNavbar && (
-                    <IconButton
-                        color="inherit"
-                        aria-label="Menu"
-                        className={styles.headerIconBtn}
-                    >
-                        <Menu />
-                    </IconButton>
-                )}
-            </Hidden>
-            {showNavbar && <Navbar />}
-        </Toolbar>
-    </AppBar>
+const HackathonTitle = () => (
+    <>
+        <Logo
+            className={styles.headerLogoImg}
+            alt="Hackathon logo"
+            data-testid="headerLogo"
+        />
+        <Typography variant="h6" data-testid="hackathonName">
+            Hackathon Name
+        </Typography>
+    </>
 );
 
+const Header = ({ showNavbar = true }) => {
+    const [mobileOpen, setMobileOpen] = React.useState(false);
+
+    const toggleMenu = () => {
+        setMobileOpen(!mobileOpen);
+    };
+
+    return (
+        <AppBar className={styles.header} position="sticky">
+            <Toolbar className={styles.headerToolbar}>
+                <Hidden implementation="css" smDown={!!showNavbar}>
+                    <div className={styles.headerLogo}>
+                        <HackathonTitle />
+                    </div>
+                </Hidden>
+                <Hidden implementation="css" mdUp>
+                    {showNavbar && (
+                        <IconButton
+                            color="inherit"
+                            aria-label="Menu"
+                            className={styles.headerIconBtn}
+                            onClick={toggleMenu}
+                            data-testid="menuBtn"
+                        >
+                            <Menu />
+                        </IconButton>
+                    )}
+                </Hidden>
+                {showNavbar && <Navbar cartQuantity={cartQuantity} />}
+
+                <Drawer
+                    anchor="left"
+                    open={mobileOpen}
+                    onClose={toggleMenu}
+                    className={styles.headerDrawer}
+                >
+                    <div className={styles.headerDrawerTop}></div>
+                    <div className={styles.headerDrawerHackathon}>
+                        <HackathonTitle />
+                        <Typography variant="body2">{userEmail}</Typography>
+                    </div>
+                    <div className={styles.headerDrawerDivider}></div>
+                    <NavbarDrawer />
```

Comment:
You should maybe have the check for `showNavbar` here. I got it to show up by setting the state default true and going to the login page, I don't think it's possible to make that happen normally though. 

The drawer unmounts when it's closed, so there isn't a performance difference.